### PR TITLE
cksum: fix error handling

### LINKED ISSUE #6796: cksum: error handling deviation from GNU
`cksum`'s behavior when encountering directories or non existing files mismatches gnu implementation

```sh
$ cat <<EOF > CHECKSUM
SHA1 (dir) = ffffffffffffffffffffffffffffffffffffffff
SHA1 (not-file) = ffffffffffffffffffffffffffffffffffffffff
EOF
$ mkdir dir

$ ../../../gnu/src/cksum -c CHECKSUM  
../../../gnu/src/cksum: dir: Is a directory
dir: FAILED open or read
../../../gnu/src/cksum: not-file: No such file or directory
not-file: FAILED open or read
../../../gnu/src/cksum: WARNING: 2 listed files could not be read

$ ../../target/debug/cksum -c CHECKSUM
../../target/debug/cksum: dir: Is a directory
../../target/debug/cksum: not-file: No such file or directory
not-file: FAILED open or read
../../target/debug/cksum: WARNING: 1 listed file could not be read

# With ignore missing

$ ../../../gnu/src/cksum -c CHECKSUM --ignore-missing
../../../gnu/src/cksum: dir: Is a directory
dir: FAILED open or read
../../../gnu/src/cksum: WARNING: 1 listed file could not be read
../../../gnu/src/cksum: CHECKSUM: no file was verified

$ ../../target/debug/cksum -c CHECKSUM --ignore-missing
../../target/debug/cksum: dir: Is a directory
../../target/debug/cksum: CHECKSUM: no file was verified
```

### PR DESCRIPTION
**Pull request by Luv-Ray**
fixes #6796

### CODE REVIEW

File: tests/by-util/test_cksum.rs [hunk 1]
```diff
@@ -1344,3 +1344,40 @@ fn test_check_comment_leading_space() {
         .stdout_contains("foo: OK")
         .stderr_contains("WARNING: 1 line is improperly formatted");
 }
+
+#[cfg(not(windows))]
```
Comment (sylvestre) [inline tests/by-util/test_cksum.rs:1348]: why not windows?

Comment (Luv-Ray) [reply tests/by-util/test_cksum.rs:1348]: https://github.com/uutils/coreutils/actions/runs/11424022765/job/31783927716#step:7:3602
seems that windows has a different logic and is guarded by another test
https://github.com/uutils/coreutils/blob/9a6f5521b9d0477192501c406c3f774fe0a0276f/tests/by-util/test_cksum.rs#L1195-L1198

Comment (sylvestre) [reply tests/by-util/test_cksum.rs:1348]: could you please document this into the test ?

Comment (Luv-Ray) [reply tests/by-util/test_cksum.rs:1350]: Done.

File: src/uucore/src/lib/features/checksum.rs [hunk 2]
```diff
@@ -612,6 +615,9 @@ where
             return Ok(());
         }
 
+        // if any incorrectly formatted line, show it
+        cksum_output(&res, status);
+
```
Comment (RenjiSann) [inline src/uucore/src/lib/features/checksum.rs:620]: Is there a specific reason for moving this up ?

File: tests/by-util/test_cksum.rs [hunk 3]
```diff
@@ -1344,3 +1344,42 @@ fn test_check_comment_leading_space() {
         .stdout_contains("foo: OK")
         .stderr_contains("WARNING: 1 line is improperly formatted");
 }
+
+/// This test checks alignment with GNU's error handling.
+/// Windows has a different logic and is guarded by [`test_check_directory_error`].
+#[cfg(not(windows))]
+#[test]
+fn test_check_error_handling() {
```
Comment (RenjiSann) [inline tests/by-util/test_cksum.rs:1352]: I'd rename this `test_check_failed_to_read`, to be a bit more specific.

**Review by RenjiSann — CHANGES_REQUESTED** [review]
Please fixup the second and third commits into the first one, for better readability after merge.
Other than that, LGTM :+1:

File: src/uucore/src/lib/features/checksum.rs (hunk 2 above)
Comment (Luv-Ray) [reply src/uucore/src/lib/features/checksum.rs:620]: As shown in the code snippet in #6796 :
```
$ ../../../gnu/src/cksum -c CHECKSUM --ignore-missing
../../../gnu/src/cksum: dir: Is a directory
dir: FAILED open or read
../../../gnu/src/cksum: WARNING: 1 listed file could not be read
../../../gnu/src/cksum: CHECKSUM: no file was verified
```
warnings in `cksum_output` show up first,  and `CHECKSUM: no file was verified` shows up later.

Comment (RenjiSann) [reply src/uucore/src/lib/features/checksum.rs:620]: Right, indeed :+1:

**Comment by cakebaker** [comment]
Thanks @Luv-Ray and @RenjiSann !

**Pull request merged**